src/Geta.Optimizely.ProductFeed.Google/GoogleProductFeedConverter.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Geta Digital. All rights reserved.
+// Licensed under Apache-2.0. See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using EPiServer.Commerce.Catalog.ContentTypes;
+using Geta.Optimizely.ProductFeed.Configuration;
+using Geta.Optimizely.ProductFeed.Google.Models;
+using Geta.Optimizely.ProductFeed.Models;
+
+namespace Geta.Optimizely.ProductFeed.Google
+{
+    public class GoogleProductFeedConverter : IProductFeedContentConverter
+    {
+        private readonly Func<Type, IProductFeedEntityMapper> _mapperFactory;
+
+        public GoogleProductFeedConverter(Func<Type, IProductFeedEntityMapper> mapperFactory)
+        {
+            _mapperFactory = mapperFactory ?? throw new ArgumentNullException(nameof(mapperFactory));
+        }
+
+        public ICollection<FeedEntity> Convert(
+            ICollection<CatalogContentBase> sourceData,
+            FeedDescriptor feedDescriptor)
+        {
+            var generatedFeeds = new List<Feed>();
+            var mapper = _mapperFactory(feedDescriptor.Mapper);
+
+            var feedEntity = mapper.GenerateFeedEntity(feedDescriptor);
+
+            if (feedEntity == null)
+            {
+                return new List<FeedEntity>();
+            }
+
+            feedEntity.Entries = sourceData
+                .Select(sd => mapper.GenerateEntry(sd))
+                .Where(sd => sd != null)
+                .ToList();
```

Comment:
Remove `ToList`, probably not the best idea to iterate here and load it all in memory